nncf/experimental/quantization/compression_builder.py

```diff
@@ -80,26 +74,25 @@ def apply(self, model: TModel, dataset: Dataset) -> TModel:
         5) Collect all statistics.
         6) Apply algorithms.
         """
-
         if not self.algorithms:
             nncf_logger.info('There are no algorithms added. The original model will be returned.')
             return model
 
-        backend = get_backend(model)
+        _model = deepcopy(model)
```

Comment:
Why is it necessary?

Reply:
The user doesn't expect that the original model that he passed will be changed, I suppose. 
We should work on a copy of that.